src/Riok.Mapperly/Descriptors/MappingBuilderContext.cs

```diff
@@ -212,6 +216,17 @@ protected MappingBuilderContext(MappingBuilderContext ctx, IMethodSymbol? userSy
     public NullFallbackValue GetNullFallbackValue(ITypeSymbol? targetType = null) =>
         GetNullFallbackValue(targetType ?? Target, MapperConfiguration.ThrowOnMappingNullMismatch);
 
+    public FormatProvider? GetFormatProvider(string? formatProviderName)
+    {
+        var formatProvider = FormatProviders.Get(formatProviderName);
+        if (formatProvider == null && formatProviderName != null)
```

Comment:
Most of the time `formatProvider` will be `null` since no default format provider is specified and `formatProviderName` will also be `null`. Then it would be faster to check first for the second condition and then the first one.

Maybe I'm wrong, or maybe the time saving is negligible.

Reply:
Switched